docs/en/cookbook/working-with-datetime.rst

```diff
@@ -67,53 +68,190 @@ The problem is simple. Not a single database vendor saves the timezone, only the
+Therefore we need to store the timezone information as well and also need to provide a way to get the datetime
+back from the database with the correct timezone-information.
+
+To be able to transform these values back into their real timezone we have to save the timezone in a separate
+field of the entity requiring timezoned datetimes:
+
+.. code-block:: php
+
+    <?php
+    namespace Calendaring;
+
+    use DateTimeImmutable;
+    use DateTimeInterface;
+    use DateTimezone;
+    use Doctrine\ORM\Mapping as ORM;
+
+    /**
+     * @ORM\Entity
+     */
+    class Event
+    {
+        /**
+         * @ORM\Id
+         * @ORM\GeneratedValue
+         * @ORM\Column(type="integer")
+         */
+        private ?int $id;
+
+        /** @ORM\Column(type="datetime") */
+        private DateTimeInterface $eventDateTime;
+
+        /** @ORM\Column(type="string") */
+        private string $timezone;
+
+        public function __construct(DateTimeInterface $eventDateTime)
+        {
+            $this->eventDateTime = $eventDateTime;
+            $this->timezone = $eventDateTime->getTimeZone()->getName();
+        }
+
+        public function getEventDateTime()
+        {
+            return new DateTimeImmutable(
+                $this->eventDateTime->format('Y-m-d H:i:s'),
+                new DateTimeZone($this->timezone)
+            );
+        }
+    }
+
+During hydration $this->eventDateTime will set with a new DateTimeImmutable instance with the servers
+default timezone. So when getting the eventDateTime we need to make sure that the correct timezone is set.
+Therefore we return a correctly created DateTimeImmutable object in the `getEventDateTime()`-method
+
+DBALTypes sadly currently can not take care of this during hydration due to the fact that they are not
+allowing to access multiple fields.
+
+An alternative would be to use a post-processing lifecycle event to recreate the object in the right way like this:
+
+.. code-block:: php
+
+    <?php
+    namespace Calendaring;
+
+    use DateTimeImmutable;
+    use DateTimeInterface;
+    use DateTimezone;
+    use Doctrine\ORM\Mapping as ORM;
+
+    /**
+     * @ORM\Entity
+     * @ORM\HasLifecycleCallbacks
+     */
+    class Event
+    {
+        /**
+         * @ORM\Id
+         * @ORM\GeneratedValue
+         * @ORM\Column(type="integer")
+         */
+        private ?int $id;
+
+        /** @ORM\Column(type="datetime") */
+        private DateTimeInterface $eventDateTime;
+
+        /** @ORM\Column(type="string") */
+        private string $timezone;
+
+        public function __construct(DateTimeInterface $eventDateTime)
+        {
+            $this->eventDateTime = $eventDateTime;
+            $this->timezone = $eventDateTime->getTimeZone()->getName();
+        }
+
+        /** @PostLoad */
+        public function correctTimezone(): void
+        {
+            $this->eventDateTime = new DateTimeImmutable(
```

Comment:
Writing a new DateTimeImmutable into `$this->eventDateTime` will trigger an update for the entities when `$entityManager->flush()` is called. Do you need this alternative `Event` entity example?

Reply:
I just did a quick check. As far as I understand [the code](https://github.com/doctrine/orm/blob/2.9.x/lib/Doctrine/ORM/UnitOfWork.php#L2941), the `PostLoad` event listeners are run as the last part of the hydration process. Therefore I expect that the entity is not marked as changed.

Reply:
Yes, having the following code in an Entity

``` php
    /** @ORM\PostLoad */
    public function fooLoad(): void
    {
        $this->bar = 'lol';
    }
```

and running 


``` php
$foo = $this->entityManager->find(\App\Entity\Foo::class, 1);
$this->entityManager->flush();
```

Will write "lol" into the mapped table column of `$this->bar`.

---

There's also a note about combined usage of `PostLoad` and `Doctrine\ORM\AbstractQuery#toIterable()` that is not considered safe.
https://www.doctrine-project.org/projects/doctrine-orm/en/2.9/reference/events.html#lifecycle-events